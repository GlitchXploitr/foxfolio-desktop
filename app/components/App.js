// @flow
import type { Children } from 'react';
import React, { Component } from 'react';
import { withStyles } from 'material-ui/styles';
import Drawer from 'material-ui/Drawer';
import { AppBar, List, ListItem, ListItemIcon, ListItemText, Toolbar, Typography } from 'material-ui';
import { Cloud, Dashboard } from 'material-ui-icons';
import { NavLink } from 'react-router-dom';
import { continuouslyFetchTransactions } from '../actions/transactions';

const drawerWidth = 250;

const styles = theme => ({
  appFrame: {
    width: '100%',
    height: '100%',
  },
  content: {
    backgroundColor: theme.palette.background.default,
    padding: theme.spacing.unit * 3,
    height: 'calc(100% - 56px)',
    marginLeft: drawerWidth,
  },
  drawerPaper: {
    width: drawerWidth,
  },
  bottomList: {
    marginTop: 'auto',
    flexGrow: 0,
  },
});

class App extends Component {
  props: {
    dispatch: (action: any) => void,
    children: Children,
    classes: any
  };

  componentDidMount() {
    this.props.dispatch(continuouslyFetchTransactions());
  }

  render() {
    const { classes, children } = this.props;
    return (
      <div className={classes.appFrame}>
        <Drawer type="permanent" classes={{ paper: classes.drawerPaper }}>
          <AppBar position="static" color="primary" elevation={1}>
            <Toolbar>
              <Typography type="title" color="inherit">
                Awesome Portfolio
              </Typography>
            </Toolbar>
          </AppBar>
          <List>
            <ListItem button component={NavLink} exact to="/">
              <ListItemIcon>
                <Dashboard/>
              </ListItemIcon>
              <ListItemText primary="Portfolio"/>
            </ListItem>
          </List>
          <List className={classes.bottomList}>
            <ListItem button component={NavLink} to="/sources">
              <ListItemIcon>
                <Cloud/>
              </ListItemIcon>
              <ListItemText primary="Sources"/>
            </ListItem>
          </List>
        </Drawer>
        <main className={classes.content}>
          {children}
        </main>
      </div>
    );
  }
}

export default withStyles(styles)(App);
