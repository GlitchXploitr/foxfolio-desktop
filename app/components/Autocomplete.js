// @flow
import React from 'react';
import Downshift from 'downshift';
import { MenuItem, Paper, TextField, withStyles } from 'material-ui';

const styles = {};

type Props = {
  id: string,
  label: string,
  classes: Object,
  onChange: (string) => void,
  value: string,
  items: string[]
};

const isSelected = (selection: string, item: string) =>
  !selection || item.toLowerCase().includes(selection.toLowerCase());

function UnstyledAutocomplete(props: Props) {
  const { classes, id, label, items, onChange, value } = props;

  return (
    <Downshift onChange={onChange} defaultSelectedItem={value}>
      {({ getInputProps, getItemProps, isOpen, inputValue, highlightedIndex }) => (
        <div>
          <TextField
            fullWidth
            autoFocus
            label={label}
            InputProps={getInputProps({ id })}
          />
          {isOpen ? (
            <Paper square className={classes.paper}>
              {items
                .filter(exchange => isSelected(inputValue, exchange))
                .slice(0, 5)
                .map((suggestion, index) => (
                  <MenuItem
                    {...getItemProps({ item: suggestion })}
                    key={suggestion}
                    selected={highlightedIndex === index}
                    component="div"
                  >
                    {suggestion}
                  </MenuItem>),
                )}
            </Paper>
          ) : null}
        </div>
      )}
    </Downshift>
  );
}

export const Autocomplete = withStyles(styles)(UnstyledAutocomplete);
