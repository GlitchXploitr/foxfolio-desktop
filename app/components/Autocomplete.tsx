import Downshift from 'downshift';
import { MenuItem, Paper, TextField, withStyles } from 'material-ui';
import { StyleRules } from 'material-ui/styles';
import React from 'react';

const styles: StyleRules = {
  container: { height: 50, position: 'relative' },
  paper: { position: 'absolute', width: '100%', zIndex: 9999 },
};

interface Props {
  id: string;
  label: string;
  onChange: (item: string) => any;
  value: string;
  items: string[];
}

const isSelected = (selection: string | null, item: string) =>
  !selection || item.toLowerCase().includes(selection.toLowerCase());

export const Autocomplete = withStyles(styles)<Props>(
  ({ classes, id, label, items, onChange, value }) => (
    <Downshift
      onChange={onChange}
      itemToString={i => (i == null ? '' : String(i))}
      defaultSelectedItem={value}
    >
      {({ getInputProps, getItemProps, isOpen, inputValue, highlightedIndex }) => (
        <div className={classes.container}>
          <TextField fullWidth autoFocus label={label} InputProps={getInputProps({ id })} />
          {isOpen ? (
            <Paper square className={classes.paper}>
              {items
                .filter(exchange => isSelected(inputValue, exchange))
                .slice(0, 5)
                .map((suggestion, index) => (
                  <MenuItem
                    {...getItemProps({ item: suggestion })}
                    key={suggestion}
                    selected={highlightedIndex === index}
                    component="div"
                  >
                    {suggestion}
                  </MenuItem>
                ))}
            </Paper>
          ) : null}
        </div>
      )}
    </Downshift>
  )
);
