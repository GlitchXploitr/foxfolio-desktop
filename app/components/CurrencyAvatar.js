// @flow
import React from 'react';
import { Avatar } from 'material-ui';
import { AccountBalance, HelpOutline } from 'material-ui-icons';
import type { Coinlist } from '../reducers/coinlist/types.d';
import { getFiatCurrencies } from '../utils/fiatCurrencies';

type Props = {
  asset: string,
  coinlist: Coinlist,
  fiatClass: Object
};

export const CurrencyAvatar = ({ asset, coinlist, fiatClass }: Props) => {
  let avatar = <Avatar><HelpOutline/></Avatar>;
  if (coinlist[asset]) {
    avatar = <Avatar src={`https://www.cryptocompare.com${coinlist[asset].ImageUrl}`}/>;
  } else if (getFiatCurrencies().includes(asset)) {
    avatar = <Avatar className={fiatClass}><AccountBalance/></Avatar>;
  }
  return avatar;
};
