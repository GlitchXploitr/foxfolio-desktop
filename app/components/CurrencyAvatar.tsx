import { Avatar } from 'material-ui';
import { AccountBalance, HelpOutline } from 'material-ui-icons';
import React from 'react';

import { Coinlist } from '../reducers/coinlist';
import { getFiatCurrencies } from '../utils/fiatCurrencies';

interface Props {
  asset: string;
  coinlist: Coinlist;
  fiatClass: string;
}

export const CurrencyAvatar = ({ asset, coinlist, fiatClass }: Props) => {
  let avatar = (
    <Avatar>
      <HelpOutline />
    </Avatar>
  );
  if (coinlist[asset]) {
    avatar = <Avatar src={`https://www.cryptocompare.com${coinlist[asset].ImageUrl}`} />;
  } else if (getFiatCurrencies().includes(asset)) {
    avatar = (
      <Avatar className={fiatClass}>
        <AccountBalance />
      </Avatar>
    );
  }
  return avatar;
};
