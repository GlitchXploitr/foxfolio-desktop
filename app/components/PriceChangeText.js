// @flow
import React from 'react';
import { withStyles } from 'material-ui';

type Props = {
  change: number,
  classes: Object
};

const styles = theme => ({
  negative: {
    color: theme.palette.error[500],
  },
  positive: {
    color: theme.palette.success[500],
  },
});

function PriceChangeText({ change, classes }: Props) {
  return (
    <span className={change > 0 ? classes.positive : classes.negative}>
      {`${change > 0 ? '+' : ''}${(change).toFixed(2)}%`}
    </span>
  );
}

export default withStyles(styles)(PriceChangeText);
