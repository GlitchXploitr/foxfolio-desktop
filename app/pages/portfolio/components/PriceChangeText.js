// @flow
import React from 'react';
import { withStyles } from 'material-ui';
import { green, grey, red } from 'material-ui/colors';

type Props = {
  change: number,
  muted: boolean,
  classes: Object
};

const styles = () => ({
  negative: {
    color: red[500],
  },
  negativeMuted: {
    color: red[300],
  },
  positive: {
    color: green[500],
  },
  positiveMuted: {
    color: green[300],
  },
  interval: {
    fontSize: 12,
    color: grey[700],
  },
  intervalMuted: {
    fontSize: 12,
    color: grey[500],
  },
});

function PriceChangeText({ change, classes, muted }: Props) {
  let className = classes.positive;
  if (change > 0 && muted) {
    className = classes.positiveMuted;
  } else if (change < 0 && !muted) {
    className = classes.negative;
  } else if (change < 0 && muted) {
    className = classes.negativeMuted;
  }
  return (
    <div style={{ display: 'inline-block' }}>
      <span className={className}>
        {`${change > 0 ? '+' : ''}${(change).toFixed(2)}%`}
      </span>
      <span className={muted ? classes.intervalMuted : classes.interval}> /24h</span>
    </div>
  );
}

export default withStyles(styles)(PriceChangeText);
