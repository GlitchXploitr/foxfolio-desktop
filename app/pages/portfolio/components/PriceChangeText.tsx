// @flow
import { withStyles } from 'material-ui';
import { green, grey, red } from 'material-ui/colors';
import React from 'react';

interface Props {
  change: number;
  muted?: boolean;
}

const styles = () => ({
  negative: {
    color: red[500],
  },
  negativeMuted: {
    color: red[300],
  },
  positive: {
    color: green[500],
  },
  positiveMuted: {
    color: green[300],
  },
  interval: {
    fontSize: 12,
    color: grey[700],
  },
  intervalMuted: {
    fontSize: 12,
    color: grey[500],
  },
});

export const PriceChangeText = withStyles(styles)<Props>(({ change, classes, muted }) => {
  let className = classes.positive;
  if (change > 0 && muted) {
    className = classes.positiveMuted;
  } else if (change < 0 && !muted) {
    className = classes.negative;
  } else if (change < 0 && muted) {
    className = classes.negativeMuted;
  }
  return (
    <div style={{ display: 'inline-block' }}>
      <span className={className}>{`${change > 0 ? '+' : ''}${change.toFixed(2)}%`}</span>
      <span className={muted ? classes.intervalMuted : classes.interval}> /24h</span>
    </div>
  );
});
