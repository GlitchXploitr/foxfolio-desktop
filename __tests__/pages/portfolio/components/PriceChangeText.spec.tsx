import { createShallow } from 'material-ui/test-utils';
import * as React from 'react';
import { PriceChangeText } from '../../../../app/pages/portfolio/components/PriceChangeText';

let shallow;

beforeAll(() => {
  shallow = createShallow();
});

test('Positive change', () => {
  const wrapper = shallow(<PriceChangeText change={5} />).dive();
  expect(wrapper).toMatchSnapshot();
});

test('Negative change', () => {
  const wrapper = shallow(<PriceChangeText change={-5} />).dive();
  expect(wrapper).toMatchSnapshot();
});
