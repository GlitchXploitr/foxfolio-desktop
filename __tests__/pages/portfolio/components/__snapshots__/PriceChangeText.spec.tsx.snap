// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Negative change 1`] = `
<div
  style={
    Object {
      "display": "inline-block",
    }
  }
>
  <span
    className="Component-negative-1"
  >
    -5.00%
  </span>
  <span
    className="Component-interval-5"
  >
     /24h
  </span>
</div>
`;

exports[`Positive change 1`] = `
<div
  style={
    Object {
      "display": "inline-block",
    }
  }
>
  <span
    className="Component-positive-3"
  >
    +5.00%
  </span>
  <span
    className="Component-interval-5"
  >
     /24h
  </span>
</div>
`;
